analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -57,9 +57,18 @@ internal sealed class FieldReference : SimpleProcessor<IFieldReferenceOperationW
 
     protected override ProgramState Process(SymbolicContext context, IFieldReferenceOperationWrapper fieldReference)
     {
-        var state = fieldReference.WrappedOperation.TrackedSymbol() is { } fieldSymbol && context.State[fieldSymbol] is { } value
-            ? context.State.SetOperationValue(context.Operation, value)
-            : context.State;
+        var state = context.State;
+        if (fieldReference.WrappedOperation.TrackedSymbol() is IFieldSymbol fieldSymbol)
+        {
+            if (context.State[fieldSymbol] is { } value)
+            {
+                state = state.SetOperationValue(context.Operation, value);
+            }
+            if (fieldSymbol.Type.IsNonNullableValueType() || fieldSymbol.Type.IsEnum())
+            {
+                state = state.SetSymbolConstraint(fieldSymbol, ObjectConstraint.NotNull);
```

Comment:
I don't think this is enough. We have this problem in general for symbols. A similar change needs to be done also for parameters and local references.
So we'll need to extract it somewhere.
As such, `ISymbol.GetSymbolType()` extension can be used to get the type, instead of the `IFieldSymbol` check.